samples/migrations/migrate_qdrant_vectorstore_to_alloydb.py

```diff
@@ -0,0 +1,185 @@
+    qdrant_batch_size: int = QDRANT_BATCH_SIZE,
+    qdrant_path: str = QDRANT_PATH,
+    project_id: str = PROJECT_ID,
+    region: str = REGION,
+    cluster: str = CLUSTER,
+    instance: str = INSTANCE,
+    alloydb_table: str = ALLOYDB_TABLE_NAME,
+    db_name: str = DB_NAME,
+    db_user: str = DB_USER,
+    db_pwd: str = DB_PWD,
+) -> None:
+    # [START qdrant_get_client]
+    from qdrant_client import QdrantClient
+
+    qdrant_client = QdrantClient(path=qdrant_path)
+
+    # [END qdrant_get_client]
+    print("Qdrant client initiated.")
+
+    # [START langchain_alloydb_migration_get_client]
+    from langchain_google_alloydb_pg import AlloyDBEngine
+
+    alloydb_engine = await AlloyDBEngine.afrom_instance(
+        project_id=project_id,
+        region=region,
+        cluster=cluster,
+        instance=instance,
+        database=db_name,
+        user=db_user,
+        password=db_pwd,
+    )
+    # [END langchain_alloydb_migration_get_client]
+    print("Langchain AlloyDB client initiated.")
+
+    # [START langchain_alloydb_migration_fake_embedding_service]
+    from langchain_core.embeddings import FakeEmbeddings
+
+    embeddings_service = FakeEmbeddings(size=qdrant_vector_size)
+    # [END langchain_alloydb_migration_fake_embedding_service]
+    print("Langchain Fake Embeddings service initiated.")
+
+    # [START qdrant_alloydb_migration_get_alloydb_vectorstore]
+
+    await alloydb_engine.ainit_vectorstore_table(
+        table_name=alloydb_table,
+        vector_size=qdrant_vector_size,
+        overwrite_existing=True,
+    )
+
+    # [START langchain_get_alloydb_migration_vector_store]
+    from langchain_google_alloydb_pg import AlloyDBVectorStore
+
+    vs = await AlloyDBVectorStore.create(
+        engine=alloydb_engine,
+        embedding_service=embeddings_service,
+        table_name=alloydb_table,
+    )
+    # [END langchain_get_alloydb_migration_vector_store]
+    print("Langchain AlloyDB vector store instantiated.")
+
+    # [END qdrant_alloydb_migration_get_alloydb_vectorstore]
+    print("Qdrant migration AlloyDBVectorStore table created.")
+
+    data_iterator = get_data_batch(
+        qdrant_client=qdrant_client,
+        qdrant_batch_size=qdrant_batch_size,
+        qdrant_collection_name=qdrant_collection_name,
+    )
+
+    # [START qdrant_alloydb_migration_insert_data_batch]
+
+    # [START langchain_alloydb_migration_vector_store_insert_data]
+    pending: set[Any] = set()
+    for ids, contents, embeddings, metadatas in data_iterator:
+        pending.add(
+            asyncio.ensure_future(
+                vs.aadd_embeddings(
+                    texts=contents,
+                    embeddings=embeddings,
+                    metadatas=metadatas,
+                    ids=ids,
+                )
+            )
+        )
+        if len(pending) >= MAX_CONCURRENCY:
+            _, pending = await asyncio.wait(
+                pending, return_when=asyncio.FIRST_COMPLETED
```

Comment:
Check whether it should be `ALL_COMPLETED` instead?

Reply:
FIRST_COMPLETED is correct, as soon as the first one is finished processing we can start processing more